src/content/blog/content-layer-deep-dive.mdx

```diff
@@ -0,0 +1,236 @@
+---
+title: 'Content Layer: A Deep Dive'
+description: There's a new way to handle content in Astro. Take a deep dive into the Content Layer API, and learn how it can help you build even more kinds of sites using Astro.
+publishDate: '2024-09-18'
+authors:
+  - matt
+lang: 'en'
+coverImage: "/src/content/blog/_images/content-layer-deep-dive/blog-post-content-layer.png"
+socialImage: "/src/content/blog/_images/content-layer-deep-dive/og-content-layer.png"
+---
+
+import BlogContentImage from '/src/components/BlogContentImage.astro';
+import astroBuildImage from '/src/content/blog/_images/content-layer-deep-dive/astro-build.png';
+import contentLayerArchitectureImage from '/src/content/blog/_images/content-layer-deep-dive/content-layer-architecture.png';
+
+**This week we have released the first beta of Astro 5, which includes a whole new way to handle content in Astro. This post takes a deep dive into the Content Layer API, showing how it works and how you can use it to build your sites.**
+
+Astro was born to create content-driven websites. While it can now be used to build all sorts of dynamic apps too, it is still the very best for sites that are built around lots of content. From full-featured docs sites built with Starlight like [Cloudflare](https://developers.cloudflare.com/) and [StackBlitz](https://tutorialkit.dev/), to beautiful marketing sites for brands such as [Porsche](https://www.porsche.com/usa/) and [Netlify](https://www.netlify.com), millions of users every day are already appreciating the fast and accessible sites that you can build with Astro, and thousands of engineers love the best developer experience in the industry.
+
+In Astro 2 we introduced [Content Collections](https://docs.astro.build/en/guides/content-collections/) as a powerful new way to organize your local content, build with type safety, and scale to thousands of pages. Content Collections provide a best-in-class developer experience for local files such as Markdown and MDX, but we heard from you that you wanted the same benefits for all of your content including remote APIs. It was also clear that while many people were building sites with thousands of pages, our Content Collections API struggled to scale into the tens of thousands of pages, with slower builds and excessive memory usage.
+
+In June [we shared an early preview of our plan](https://astro.build/blog/future-of-astro-content-layer/) to solve these problems and more with an all new type of content collection backed by an all new Content Layer API that gives you the flexibility you asked for. We imagined taking collections beyond just files in `src/content`, letting you load your content from anywhere. We have been testing with collections that scale up to a size never before possible. From our [first experimental release in Astro 4.14](/blog/astro-4140/#experimental-content-layer-api), we have been hard at work stabilizing this new API for release in Astro 5.
```

Comment:
I am not sure about the repeating "all new"? Also, it's missing hyphens in this case. I would change it slightly so we don't repeat it.
```suggestion
In June [we shared an early preview of our plan](https://astro.build/blog/future-of-astro-content-layer/) to solve these problems and more with an all-new type of content collection backed by the Content Layer API that gives you the flexibility you asked for. We imagined taking collections beyond just files in `src/content`, letting you load your content from anywhere. We have been testing with collections that scale up to a size never before possible. From our [first experimental release in Astro 4.14](/blog/astro-4140/#experimental-content-layer-api), we have been hard at work stabilizing this new API for release in Astro 5.
```